### PR TITLE
kernel-hci: Update config to enable DM multipath Kernel configurations.

### PR DESCRIPTION
**Pull request by liulanze**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

###### Change Log  

- Add patch (0028) to enable DM multipath Kernel configurations.

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id (official): 404878
- Pipeline build id (hci): 404881

### CODE REVIEW

File: SPECS/kernel-hci/0029-enable-dm-multipath-kernel-configs.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
+--- a/config	2023-08-07 14:00:37.409036062 -0700
```
Comment (cwize1) [inline SPECS/kernel-hci/0029-enable-dm-multipath-kernel-configs.patch:1]: Why not just change the `config` file directly instead of using a patch file?

**Review by cwize1 — CHANGES_REQUESTED** [review]

File: SPECS/kernel-hci/kernel-hci.spec [hunk 2]
```diff
@@ -55,6 +55,7 @@ Patch24:        0025-net-mlx5-Bridge-implement-infrastructure-for-VLAN-pr.patch
 Patch25:        0026-net-mlx5-Bridge-implement-QinQ-support.patch
 Patch26:        0027-mstflint-This-driver-enables-under-the-secure-boot.patch
 Patch27:        0028-net-mlx5-Bridge-use-debug-not-warn-if-entry-not-found.patch
+Patch28:        0029-enable-dm-multipath-kernel-configs.patch
```
Comment (vinceaperri) [inline SPECS/kernel-hci/kernel-hci.spec:58]: This is for patching the kernel-hci source code. The config file is part of the source code of this repository, at SPECS/kernel-hci/config. So you can just make the desired changes there.

File: SPECS/kernel-hci/kernel-hci.spec [hunk 3]
```diff
@@ -431,6 +433,9 @@ ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
 %{_sysconfdir}/bash_completion.d/bpftool
 
 %changelog
+* Mon Aug 07 2023 Lanze Liu <lanzeliu@microsoft.com> - 5.15.123.1-2
+- Add patch (0028) to enable DM multipath Kernel configurations
```
Comment (vinceaperri) [inline SPECS/kernel-hci/kernel-hci.spec:437]: Please specify the specific CONFIG options are are enabling/changing/disabling here.

File: SPECS/kernel-hci/kernel-hci.spec [hunk 4]
```diff
@@ -198,6 +199,7 @@ manipulation of eBPF programs and maps.
 %patch25 -p1
 %patch26 -p1
 %patch27 -p1
+%patch28 -p1
```
Comment (vinceaperri) [inline SPECS/kernel-hci/kernel-hci.spec:202]: Not needed after you remove patch.

**Review by vinceaperri — CHANGES_REQUESTED** [review]

**Review by cwize1 — COMMENTED** [review]
LGTM

**Review by vinceaperri — APPROVED** [review]

**Review by cwize1 — APPROVED** [review]

**Pull request merged**